### PR TITLE
WebSocket implementation zio-quickstart.g8

### PR DESCRIPTION
**Pull request by ernestochero**

### CODE REVIEW

File: src/main/g8/src/main/scala/$package$/service/$if(add_websocket_endpoint.truthy)$SubscriberService.scala$endif$ [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.example.service
```
Comment (sviezypan) [inline src/main/g8/src/main/scala/$package$/service/$if(add_websocket_endpoint.truthy)$SubscriberService.scala$endif$:1]: this one needs to be package $package$.service

File: src/main/g8/src/main/scala/$package$/service/$if(add_websocket_endpoint.truthy)$SubscriberService.scala$endif$ [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import $package$.domain.ItemId
+import zio.*
+import zio.stream.*
+
+trait SubscriberService:
+  def publishDeleteEvents(deletedItemId: ItemId): IO[Nothing, List[Boolean]]
+  def showDeleteEvents: Stream[Nothing, ItemId]
+  def getQueue: UIO[List[Queue[ItemId]]]
+  
+object SubscriberService {
```
Comment (sviezypan) [inline src/main/g8/src/main/scala/$package$/service/$if(add_websocket_endpoint.truthy)$SubscriberService.scala$endif$:12]: lets use optional braces syntax so we're consistent across codebase :)

File: src/main/g8/src/main/scala/$package$/service/$if(add_websocket_endpoint.truthy)$SubscriberServiceLive.scala$endif$ [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package com.example.service
```
Comment (sviezypan) [inline src/main/g8/src/main/scala/$package$/service/$if(add_websocket_endpoint.truthy)$SubscriberServiceLive.scala$endif$:1]: also $package$.service, real package name will be specified by user

File: src/main/g8/src/main/scala/$package$/service/$if(add_websocket_endpoint.truthy)$SubscriberServiceLive.scala$endif$ [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package com.example.service
+
+import $package$.domain.ItemId
+import zio.*
+import zio.stream.*
+
+final case class SubscriberServiceLive(deletedEventsSubscribers: Ref[List[Queue[ItemId]]])
```
Comment (sviezypan) [inline src/main/g8/src/main/scala/$package$/service/$if(add_websocket_endpoint.truthy)$SubscriberServiceLive.scala$endif$:7]: lets use optional braces here and in companion object as well

File: src/main/g8/src/test/scala/$package$/api/$if(add_websocket_endpoint.truthy)$WebSocketRouteSpec.scala$endif$ [hunk 5]
```diff
@@ -0,0 +1,42 @@
+package com.example.api
```
Comment (sviezypan) [inline src/main/g8/src/test/scala/$package$/api/$if(add_websocket_endpoint.truthy)$WebSocketRouteSpec.scala$endif$:1]: $package$.api + optional braces syntax in this file

File: src/main/g8/src/test/scala/$package$/service/$if(add_websocket_endpoint.truthy)$SubscriberSpec.scala$endif$ [hunk 6]
```diff
@@ -0,0 +1,52 @@
+package com.example.service
```
Comment (sviezypan) [inline src/main/g8/src/test/scala/$package$/service/$if(add_websocket_endpoint.truthy)$SubscriberSpec.scala$endif$:1]: $package$.api + optional braces syntax in this file

**Review by sviezypan — CHANGES_REQUESTED** [review]
hi @ernestochero, I added few minor comments. 
One way to verify that template is fine is to "cd" into dir where zio-quickstart is cloned and do
g8 file://zio-quickstart.g8
Then you can see if generated projects builds & runs.

**Review by sviezypan — APPROVED** [review]
Looks good to me!

File: src/main/g8/src/main/scala/$package$/service/$if(add_websocket_endpoint.truthy)$SubscriberServiceLive.scala$endif$ [hunk 7]
```diff
@@ -0,0 +1,29 @@
+package $package$.service
+
+import $package$.domain.ItemId
+import zio.*
+import zio.stream.*
```
Comment (jczuchnowski) [inline src/main/g8/src/main/scala/$package$/service/$if(add_websocket_endpoint.truthy)$SubscriberServiceLive.scala$endif$:5]: `_`

File: src/main/g8/src/main/scala/$package$/service/$if(add_websocket_endpoint.truthy)$SubscriberServiceLive.scala$endif$ [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    deletedEventsSubscribers.get.flatMap { subs =>
+      UIO.foreach(subs) { queue =>
+        queue
+          .offer(deletedItemId)
+          .onInterrupt(
+            deletedEventsSubscribers.update(_.filterNot(_ == queue))
+          )
+      }
+    }
+  override def showDeleteEvents: Stream[Nothing, ItemId] = ZStream.unwrap {
```
Comment (jczuchnowski) [inline src/main/g8/src/main/scala/$package$/service/$if(add_websocket_endpoint.truthy)$SubscriberServiceLive.scala$endif$:20]: add newline between methods

**Review by jczuchnowski — APPROVED** [review]

**Pull request merged**